### PR TITLE
Nginx Hello World

### PR DESCRIPTION
**Pull request by AnsisMalins**
This changes the template project to a Nginx Hello World application that demonstrates all the basics.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -3,13 +3,103 @@ terraform {
...
+  spec {
+    port {
+      name = "http"
+      port = 80
+      target_port = 80
+    }
+    selector = {
+      k8s_app = local.app_name
+    }
+    type = "NodePort"
```
Comment (mraerino) [inline main.tf:71]: doesn't need to be exposed on a port, let's not have this in the example

File: main.tf [hunk 2]
```diff
@@ -3,13 +3,103 @@ terraform {
...
+# If applied as is, this template would put up an empty Nginx site at https://k8s-template.dev.dorf.world/
+
+locals {
+  app_name = "k8s-template" # Change to your-thing
+  host = "k8s-template.dev.dorf.world" # Change to your-thing.dev.dorf.world
+}
+
+resource "kubernetes_namespace" "app" {
+  metadata {
+    name = "app-k8s-template"
```
Comment (nomaster) [inline main.tf:20]:
```suggestion
    name = local.app_name
```

File: main.tf (hunk 1 above)
Comment (AnsisMalins) [reply main.tf:71]: You mean just get rid of that one line?

**Review by nomaster — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (nomaster) [inline main.tf:71]:
```suggestion
    type = "ClusterIP"
```

Comment (nomaster) [reply main.tf:71]:
```suggestion
    type = "ClusterIP"
```

**Review by mraerino — APPROVED** [review]

**Review by nomaster — APPROVED** [review]

**Review by mraerino — APPROVED** [review]

**Pull request merged**